### PR TITLE
fixing data_utils padding code

### PR DESCRIPTION
**Pull request by sourabh2k15**

### CODE REVIEW

File: algorithmic_efficiency/data_utils.py [hunk 1]
```diff
@@ -23,14 +23,14 @@ def shard_and_maybe_pad_np(
   create the corresponding mask, and reshape it to be sharded across devices.
   """
   local_device_count = max(torch.cuda.device_count(), jax.local_device_count())
-  current_batch_size = batch['inputs'].shape[0]
+  current_batch_size = batch['inputs'][0].shape[0]
```
Comment (runame) [inline algorithmic_efficiency/data_utils.py:26]: Ah right, librispeech has a tuple in `batch['inputs']`. This should be something like
```python
inputs = batch['inputs']
current_batch_size = inputs[0].shape[0] if isinstance(inputs, tuple) else inputs.shape[0]
```

File: algorithmic_efficiency/data_utils.py [hunk 2]
```diff
@@ -23,14 +23,14 @@ def shard_and_maybe_pad_np(
...
   remainder_size = current_batch_size % local_device_count
   if remainder_size != 0:
     if global_batch_size is not None:
       pad_size = global_batch_size - current_batch_size
     else:
       pad_size = local_device_count - remainder_size
-    targets_shape = tuple(batch['targets'].shape)
+    targets_shape = tuple(batch['targets'][0].shape)
```
Comment (runame) [inline algorithmic_efficiency/data_utils.py:33]: Same as above.

**Review by runame — APPROVED** [review]

File: algorithmic_efficiency/data_utils.py (hunk 1 above)
Comment (znado) [reply algorithmic_efficiency/data_utils.py:26]: +1 we need to make sure this works for all workloads

**Review by znado — CHANGES_REQUESTED** [review]

File: algorithmic_efficiency/data_utils.py (hunk 1 above)
Comment (sourabh2k15) [reply algorithmic_efficiency/data_utils.py]: made the changes

File: algorithmic_efficiency/data_utils.py (hunk 2 above)
Comment (sourabh2k15) [reply algorithmic_efficiency/data_utils.py]: done

**Review by znado — APPROVED** [review]

File: algorithmic_efficiency/data_utils.py (hunk 1 above)
Comment (runame) [reply algorithmic_efficiency/data_utils.py]: I think I tested for all workloads, but librispeech and Criteo1TB in the original PR (didn't have the data ready).

Comment (runame) [reply algorithmic_efficiency/data_utils.py]: I think I tested this for all workloads, but librispeech and Criteo1TB in the original PR (didn't have the data ready).
